lib/collection/src/collection/snapshots.rs

```diff
@@ -155,6 +155,7 @@ impl Collection {
         log::debug!("Archiving snapshot {snapshot_temp_target_dir_path:?}");
         let archiving = tokio::task::spawn_blocking(move || -> CollectionResult<_> {
             let mut builder = tar::Builder::new(snapshot_temp_arc_file.as_file_mut());
+            builder.sparse(true);
```

Comment:
You currently enable this everywhere. Does it make sense to gate this on non-Linux or non-Unix platforms?

Reply:
It will be no-op on unsupported platforms. And it's for packing only (unpacking sparse archives should succeed everywhere).

Also, `.sparse(true);` is redundant in the current implementation (it's enabled by default), but I decided to explicitly mark places where it is used it if we change defaults or the interface.